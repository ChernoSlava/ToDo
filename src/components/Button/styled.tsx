import styled from 'styled-components';

export const ButtonStyled = styled.div<{
  isDisabled: boolean;
}>`
  cursor: pointer;
  display: inline-block;

  ${props =>
    props.isDisabled &&
    `
    pointer-events: none;
    opacity: 0.4;
  `}
`;
