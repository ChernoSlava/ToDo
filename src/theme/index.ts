export { theme, ThemeType } from './theme';
