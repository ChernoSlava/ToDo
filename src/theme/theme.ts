export const theme = {
  colors: {
    black: 'black',
    white: '#ffffff',
    grey: 'lightgrey',
  },
};

export type ThemeType = typeof theme;
