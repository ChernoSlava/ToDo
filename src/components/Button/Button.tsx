import React, { SVGProps } from 'react';

import {
  AddIcon,
  FinishIcon,
  RemoveIcon,
  EditIcon,
  RevertIcon,
} from './images';
import { ButtonType } from './types';
import { ButtonStyled } from './styled';

// type - add, remove, edit, done, revert
const images: {
  [key in ButtonType]: React.FC<SVGProps<SVGSVGElement>>;
} = {
  add: AddIcon,
  finish: FinishIcon,
  remove: RemoveIcon,
  edit: EditIcon,
  revert: RevertIcon,
};

export const Button: React.FC<{
  title: string;
  type: ButtonType;
  onClick: () => void;
  isDisabled?: boolean;
}> = ({ title, type, onClick, isDisabled }): JSX.Element => {
  const Icon = images[type] || AddIcon;

  return (
    <ButtonStyled
      title={title}
      onClick={onClick}
      onKeyPress={onClick}
      role="button"
      aria-hidden
      isDisabled={!!isDisabled}>
      <Icon />
    </ButtonStyled>
  );
};

Button.defaultProps = {
  isDisabled: undefined,
};

export { ButtonType };
