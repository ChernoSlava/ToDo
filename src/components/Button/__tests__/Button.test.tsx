// eslint-disable import/no-extraneous-dependencies /
// eslint-disable no-undef /

import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { theme } from '../../../theme';
import { Button, ButtonType } from '../Button';

const getUi = (type: ButtonType, onClick?: () => void): JSX.Element => {
  return (
    <ThemeProvider theme={theme}>
      <Button title="I am button" onClick={onClick} type={type} />
    </ThemeProvider>
  );
};

describe('Button tests', () => {
  it('should render correct add button', () => {
    const { container } = render(getUi('add'));

    expect(container).toMatchSnapshot();
  });
});