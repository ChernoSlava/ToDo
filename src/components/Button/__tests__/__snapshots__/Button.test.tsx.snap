// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests should render correct add button 1`] = `
.c0 {
  cursor: pointer;
  display: inline-block;
}

<div>
  <div
    aria-hidden="true"
    class="c0"
    role="button"
    title="I am button"
  >
    <svg
      fill="none"
      height="25px"
      viewBox="0 0 25 25"
      width="25px"
    >
      <path
        d="M12.5 5V20M5 12.5H20"
        stroke="#121923"
        stroke-width="1.2"
      />
    </svg>
  </div>
</div>
`;
